File: CLO835_summer2026_assignment2-master/apps/go/main.go
// CLO835 Assignment 2 — Go 1.26 starter
// This is VERSION 0.2. To release VERSION 0.3, change message below to:
//   "Hello world from the CLO835 class and <YOUR_STUDENT_ID>!"
// (replace <YOUR_STUDENT_ID> with your own Seneca student ID, e.g. 10112233)
package main

import (
	"fmt"
	"log"
	"net/http"
)

const (
	version = "0.2"
	message = "Hello world from the CLO835 class!"
	port    = ":8080"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		fmt.Fprintln(w, message)
	})
	// for the bonus probes
	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) { w.WriteHeader(http.StatusOK) })
	http.HandleFunc("/readyz", func(w http.ResponseWriter, r *http.Request) { w.WriteHeader(http.StatusOK) })

	log.Printf("CLO835 app v%s listening on %s", version, port)
	log.Fatal(http.ListenAndServe(port, nil))
}

File: CLO835_summer2026_assignment2-master/apps/java/Main.java
// CLO835 Assignment 2 — Java 26 starter (JDK built-in HTTP server, no dependencies)
// This is VERSION 0.2. To release VERSION 0.3, change MESSAGE below to:
//   "Hello world from the CLO835 class and <YOUR_STUDENT_ID>!"
// (replace <YOUR_STUDENT_ID> with your own Seneca student ID, e.g. 10112233)
import com.sun.net.httpserver.HttpServer;

import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.nio.charset.StandardCharsets;

public class Main {
    static final String VERSION = "0.2";
    static final String MESSAGE = "Hello world from the CLO835 class!";

    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new InetSocketAddress("0.0.0.0", 8080), 0);
        server.createContext("/", exchange -> {
            String path = exchange.getRequestURI().getPath();
            byte[] body = (path.equals("/healthz") || path.equals("/readyz"))
                    ? "ok".getBytes(StandardCharsets.UTF_8)
                    : (MESSAGE + "\n").getBytes(StandardCharsets.UTF_8);
            exchange.sendResponseHeaders(200, body.length);
            try (OutputStream os = exchange.getResponseBody()) {
                os.write(body);
            }
        });
        System.out.println("CLO835 app v" + VERSION + " listening on :8080");
        server.start();
    }
}
